storage_service/locations/models/swift.py

```diff
@@ -0,0 +1,167 @@
+# stdlib, alphabetical
+import logging
+import os
+
+# Core Django, alphabetical
+from django.db import models
+
+# Third party dependencies, alphabetical
+import swiftclient
+
+# This project, alphabetical
+from common import utils
+
+# This module, alphabetical
+from . import StorageException
+from location import Location
+
+LOGGER = logging.getLogger(__name__)
+
+
+class Swift(models.Model):
+    space = models.OneToOneField('Space', to_field='uuid')
+    auth_url = models.CharField(max_length=256,
+        help_text='URL to authenticate against')
+    auth_version = models.CharField(max_length=8, default='2',
+        help_text='OpenStack auth version')
+    username = models.CharField(max_length=64,
+        help_text='Username to authenticate as')
+    # HELP how do I store the password?  Has to be plaintext to send to Swift, but that seems like a bad idea
```

Comment:
Not immediately practical since this isn't being executed in the context of a user, but an interesting idea: http://stackoverflow.com/a/8036765 Use a user's password as the encryption key for the password, so the password never has to be stored in plaintext, while making it decodeable.